src/NuGet.Core/NuGet.Packaging/Signing/Utility/RepositorySignatureInfoUtility.cs

```diff
@@ -51,53 +46,36 @@ public static class RepositorySignatureInfoUtility
                     fallbackSettings.AllowNoTimestamp,
                     fallbackSettings.AllowUnknownRevocation,
                     fallbackSettings.ReportUnknownRevocation,
-                    allowNoRepositoryCertificateList,
-                    fallbackSettings.AllowNoClientCertificateList,
                     fallbackSettings.VerificationTarget,
                     fallbackSettings.SignaturePlacement,
                     fallbackSettings.RepositoryCountersignatureVerificationBehavior,
-                    fallbackSettings.RevocationMode,
-                    repositoryAllowList?.AsReadOnly(),
-                    fallbackSettings.ClientCertificateList);
+                    fallbackSettings.RevocationMode);
             }
         }
 
-        private static List<CertificateHashAllowListEntry> GetRepositoryAllowList(IEnumerable<IRepositoryCertificateInfo> repositoryCertificateInfos)
+        public static IReadOnlyCollection<CertificateHashAllowListEntry> GetRepositoryAllowList(IEnumerable<IRepositoryCertificateInfo> repositoryCertificateInfos)
         {
-            List<CertificateHashAllowListEntry> repositoryAllowList = null;
+            HashSet<CertificateHashAllowListEntry> repositoryAllowList = null;
 
             if (repositoryCertificateInfos != null)
             {
-                repositoryAllowList = new List<CertificateHashAllowListEntry>();
+                repositoryAllowList = new HashSet<CertificateHashAllowListEntry>();
```

Comment:
I'm curious, why did you decide to change this to a hash set? 
Do we need deduplication logic? Is this modified anywhere else?
How many elements would the collection normally have?

Reply:
nit: also list <> set :) 

But ideally the type shouldn't be part of the name unless absolutely necessary.

Reply:
I changed this to a set because it seems unnecessary to have duplicate values in the allow list (it will only make the verification for that entry happen twice), the deduplication logic will use the [Equals]()https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/Signing/Verification/CertificateHashAllowListEntry.cs#L56) method from `CertificateHashAllowListEntry` which is sufficient.


This collection is not modified anywhere, it is the collection of certificates that a given server sends for us to verify against the package we just downloaded. 

This collection is unbounded in size (although typically it should be fairly small because it means that the server uses those certificates to sign their packages). This is why I thought it might be a good idea to dedup using a set.